java/src/jmri/implementation/MatrixSignalMast.java

```diff
@@ -374,16 +374,16 @@ public void updateOutputs(char[] bits) {
             log.debug("Empty char[] received");
         } else {
             for (int i = 0; i < outputsToBeans.size(); i++) {
-                // log.debug("Setting bits[1] = {} for output #{}", bits[i], i);
+                log.debug("Setting bits[1] = {} for output #{}", bits[i], i);
                 Turnout t = getOutputBean(i + 1);
                 if (t != null) {
                     t.setBinaryOutput(true); // prevent feedback etc.
                 }
                 if (bits[i] == '1' && t != null && t.getCommandedState() != Turnout.CLOSED) {
                     // no need to set a state already set
-                    t.setCommandedState(Turnout.CLOSED);
+                    t.setCommandedStateAtInterval(Turnout.CLOSED); // delayed on specific connection by its turnoutManager
```

Comment:
The delay here needs to be per signal mast.  Different masts on the same layout may need different delays.

Reply:
See identical comment/answer in line 286